src/send/SelectRecipientButtons.tsx

```diff
@@ -0,0 +1,177 @@
+import React, { useState } from 'react'
+import { useAsync } from 'react-async-hook'
+import { useTranslation } from 'react-i18next'
+import { Platform } from 'react-native'
+import { PERMISSIONS, PermissionStatus, RESULTS, check, request } from 'react-native-permissions'
+import { SendEvents } from 'src/analytics/Events'
+import ValoraAnalytics from 'src/analytics/ValoraAnalytics'
+import { phoneNumberVerifiedSelector } from 'src/app/selectors'
+import Dialog from 'src/components/Dialog'
+import SelectRecipientButton from 'src/components/SelectRecipientButton'
+import QRCode from 'src/icons/QRCode'
+import Social from 'src/icons/Social'
+import { navigate } from 'src/navigator/NavigationService'
+import { Screens } from 'src/navigator/Screens'
+import useSelector from 'src/redux/useSelector'
+import Logger from 'src/utils/Logger'
+import { navigateToPhoneSettings } from 'src/utils/linking'
+
+const CONTACTS_PERMISSION =
+  Platform.OS === 'ios' ? PERMISSIONS.IOS.CONTACTS : PERMISSIONS.ANDROID.READ_CONTACTS
+
+type Props = {
+  onContactsPermissionGranted: () => void
+}
+
+export default function SelectRecipientButtons({ onContactsPermissionGranted }: Props) {
+  const { t } = useTranslation()
+  const phoneNumberVerified = useSelector(phoneNumberVerifiedSelector)
+  const [contactsPermissionStatus, setContactsPermissionStatus] = useState<
+    PermissionStatus | undefined
+  >(undefined)
+  const [showConnectPhoneNumberModal, setShowConnectPhoneNumberModal] = useState(false)
+  const [showEnableContactsModal, setShowEnableContactsModal] = useState(false)
+  const [navigateToPhoneVerification, setNavigateToPhoneVerification] = useState(false)
+
+  useAsync(async () => {
+    if (!contactsPermissionStatus) {
+      setContactsPermissionStatus(await check(CONTACTS_PERMISSION))
+    }
+  }, [contactsPermissionStatus])
+
+  const onPressContacts = async () => {
+    const currentPermission = await check(CONTACTS_PERMISSION)
+    setContactsPermissionStatus(currentPermission)
```

Comment:
```suggestion
  const contactsPermissionStatus = useAsync(async () => {
    await check(CONTACTS_PERMISSION)
  }, [])

  const onPressContacts = async () => {
    const currentPermission = contactsPermissionStatus.result ?? await check(CONTACTS_PERMISSION)
```
wondering if we could do something like this to drop one state variable and avoid the useAsync loop (where state is updated that is also in the params list, which causes the hook to run again). not a big deal though